test/function_compilation.jl

```diff
@@ -0,0 +1,191 @@
+module TestFunctionCompilation
+
+using Test
+using DataFramesMeta
+
+@testset "function_compilation" begin
+	# Lazy way of making sure all
+	# functions are pre-compiled. @eval
+	# prevents julia from caching the
+	# intermediate anonymous functions.
+	for _ in 1:2
```

Comment:
Why run this twice?

Reply:
`transform(df, :x => (t -> some_stuff(t)) => :y)` should re-compiled every time, so the 2nd time should also be slow, and I wanted to ensure that.